src/Parser/Core/Modules/DamageTaken.js

```diff
@@ -20,13 +20,23 @@ class DamageTaken extends Module {
   get total() {
     return this._total;
   }
+
   _byAbility = {};
   byAbility(spellId) {
-    return this._byAbility[spellId];
+    if(!this._byAbility[spellId]) {
+      return new DamageValue(0, 0, 0);
+    } else {
```

Comment:
This else can be omitted since the first if-statement returns and functionally it works like an exception to the normal flow so for max readability I feel it's a little bit better. 😅

Reply:
I know the else isn't needed, but stylistically I prefer it. When a function is strictly "one thing or the other thing" I like the 'else' because it makes it obvious at a glance that both lines can't happen. Just a preference thing I think.